src/main_dpdk.cpp

```diff
@@ -4652,6 +4652,20 @@ COLD_FUNC void CGlobalTRex::update_stats(){
 }
 
 COLD_FUNC tx_per_flow_t CGlobalTRex::get_flow_tx_stats(uint8_t port, uint16_t index) {
+    uint8_t port0;
+    CFlowGenListPerThread * lpt;
+
+    m_stats.m_port[port].m_tx_per_flow[index].clear();
```

Comment:
You can't clear the counters while the traffic is running ..

Reply:
No, it must be cleared because it is used to accumulate the saved counters from DP threads.

Reply:
@jsmoon the code work in diff. m_tx_per_flow should go up and accumulate diff from the per port counter. 
the code that you did you did a "clear" to the counters and it does not work as it worked before ..

```
    return m_stats.m_port[port].m_tx_per_flow[index] - m_stats.m_port[port].m_prev_tx_per_flow[index];
```

the ```prev_tx_per_flow``` is updated in clear() 

The interator over the DP cores is done one every 500 msec. You can add an explicit update in this function but you can't clean the counters before

Reply:
@hhaim, frankly speaking, I had missed the clear during my implementation., and I observed the tx counter fluctuation at TUI. After I added this `clear` routine, the fluctuation disappeared.

`CGlobalTRex::get_stats` updates the counters [here](https://github.com/cisco-system-traffic-generator/trex-core/blob/8916531cc1a3a829dd821449d76061e713eea0fe/src/main_dpdk.cpp#L4822) every 500 msec and it also calls the clear [here](https://github.com/cisco-system-traffic-generator/trex-core/blob/8916531cc1a3a829dd821449d76061e713eea0fe/src/main_dpdk.cpp#L4749) for the update.